### ISSUE TITLE
Travis build credits

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#3377]
Travis currently says:

"Builds have been temporarily disabled for public repositories due to a negative credit balance. Please go to the Plan page to replenish your credit balance or alter your Consume paid credits for OSS setting."

https://travis-ci.com/github/OpenRefine/OpenRefine

I think we need to shrink our build matrix, especially if we want to add new jobs for the UX tests. I would start by removing builds marked as allowed failures.

Concerning the Java versions, I would only test with the oldest and newest versions we support.

We could also consider migrating to another provider, or requesting a raise of our usage limit, but I think it generally makes sense to try to keep the build matrix minimal, for the sake of getting a quicker feedback on PRs and saving resources.

**Comment by tfmorris** [comment]
I can't access either of the links in that warning. Do we know whether the
credits lasted most of November or expired on the first day of December or
something else?

I suspect that we're going to need to migrate away from Travis. That seems
to be what many other OSS organizations are doing, but I'm not sure what
are spend is relative to their budget or how free they are about increasing
their allowances.

The build matrix for PRs is already pretty minimal -- just Java 8, as I
recall. It's the merges that trigger the full build matrix, but we can
certainly pare that back and try to remember to manually run whatever
matrix we decide on before releases.

Tom

**Comment by wetneb** [comment]
Both links in the warning also give errors for me.

**Comment by tfmorris** [comment]
I logged in again via Github and was able to get to https://travis-ci.com/organizations/OpenRefine/plan
which says "-2410 available credits (purchase date: December 01, 2020) You have used 12410 of 10000 credits"

Implying that we used 12k credits in a few hours (or perhaps that we used that amount since Nov. 2 when they announced the pricing change). Either way, the credit was a one-time thing, not a monthly allotment, so they are gone. We can apply for more, but that would also be a one-time thing, at least by default. I know another org got 50K relatively easily (although they assigned it to the requestors account not the OSS org), so perhaps we should make the request to bridge us over while we figure out where to move.

The charge a premium for Mac OS X, so we should remove those from the matrix.
